### ISSUE TITLE
Style/GuardClause cop outputs an error

### ISSUE THREAD

**Issue by shunichi** [rubocop/rubocop#11295]
## Expected behavior

No errors occur.

## Actual behavior

Style/GuardClause cop outputs an error.

## Steps to reproduce the problem

foo.rb:
```ruby
# frozen_string_literal: true

def build_foo(user, bar)
  if (foo = user.foo)
    foo.build(
      user.bar(
        f: foo,
        b: bar
      )
    )
  end
end
```

```
$ rubocop --debug
For /home/shun/dev/src/github.com/shunichi/rubocop-bug: Default configuration from /home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/shun/dev/src/github.com/shunichi/rubocop-bug/foo.rb
An error occurred while Style/GuardClause cop was inspecting /home/shun/dev/src/github.com/shunichi/rubocop-bug/foo.rb:3:0.
undefined method `heredoc?' for s(:send,
  s(:lvar, :user), :bar,
  s(:hash,
    s(:pair,
      s(:sym, :f),
      s(:lvar, :foo)),
    s(:pair,
      s(:sym, :b),
      s(:lvar, :bar)))):RuboCop::AST::SendNode

          if if_branch&.send_type? && if_branch.last_argument&.heredoc?
                                                             ^^^^^^^^^^
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:190:in `autocorrect'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:179:in `block in register_offense'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/base.rb:346:in `correct'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/base.rb:127:in `add_offense'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:176:in `register_offense'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:149:in `check_ending_if'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:137:in `check_ending_body'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/style/guard_clause.rb:109:in `on_def'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:100:in `public_send'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:100:in `block (2 levels) in trigger_responding_cops'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:99:in `block in trigger_responding_cops'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:98:in `each'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:98:in `trigger_responding_cops'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:69:in `on_def'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.24.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:154:in `investigate_partial'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cop/team.rb:82:in `investigate'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:315:in `inspect_file'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `loop'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:138:in `block in file_offenses'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:163:in `file_offense_cache'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:137:in `file_offenses'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:128:in `process_file'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:109:in `block in each_inspected_file'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `each'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `reduce'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:108:in `each_inspected_file'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:94:in `inspect_files'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/runner.rb:47:in `run'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/command.rb:11:in `run'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli/environment.rb:18:in `run'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:72:in `run_command'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:79:in `execute_runners'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/lib/rubocop/cli.rb:48:in `run'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/exe/rubocop:19:in `block in <top (required)>'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/rubocop-1.40.0/exe/rubocop:19:in `<top (required)>'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/bin/rubocop:25:in `load'
/home/shun/.anyenv/envs/rbenv/versions/3.1.3/bin/rubocop:25:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/GuardClause cop was inspecting /home/shun/dev/src/github.com/shunichi/rubocop-bug/foo.rb:3:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.1.3) [x86_64-linux]
Finished in 0.14756669900089037 seconds
```

## RuboCop version

```
$ rubocop -V
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.1.3) [x86_64-linux]
```

**Comment by koic** [comment]
This issue is a dup and it has been solved by https://github.com/rubocop/rubocop/pull/11250.
